repositories/geometry2.BUILD.bazel

```diff
@@ -0,0 +1,65 @@
+load(
+    "@com_github_mvukov_rules_ros2//ros2:cc_defs.bzl",
+    "ros2_cpp_library",
+)
+load(
+    "@com_github_mvukov_rules_ros2//ros2:interfaces.bzl",
+    "cpp_ros2_interface_library",
+    "ros2_interface_library",
+)
+
+ros2_interface_library(
+    name = "tf2_msgs",
+    srcs = glob(["tf2_msgs/msg/*.msg", "tf2_msgs/srv/*.srv", "tf2_msgs/action/*.action",]),
+    visibility = ["//visibility:public"],
+    deps = [
+        "@ros2_rcl_interfaces//:builtin_interfaces",
+        "@ros2_common_interfaces//:geometry_msgs",
+        "@ros2_rcl_interfaces//:action_msgs",
+        "@ros2_unique_identifier_msgs//:unique_identifier_msgs",
+    ]
+)
+
+cpp_ros2_interface_library(
+    name = "cpp_tf2_msgs",
+    visibility = ["//visibility:public"],
+    deps = [":tf2_msgs",
+    ],
+)
+
+ros2_cpp_library(
+    name = "tf2",
+    srcs = glob([
+        "tf2/src/*.cpp",
+        "tf2/src/*.hpp",
+    ]),
+    hdrs = glob(["tf2/include/**/*.h"]),
+    includes = ["tf2/include"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@console_bridge",
+        "@ros2_rcutils//:rcutils",
+        "@ros2_common_interfaces//:cpp_geometry_msgs",
+    ],
+)
+
+ros2_cpp_library(
+    name = "tf2_ros",
+    srcs = glob([
+        "tf2_ros/src/*.cpp",
+        "tf2_ros/src/*.hpp",
+    ]),
+    hdrs = glob([
+        "tf2_ros/include/**/*.h",
+        "tf2_ros/include/**/*.hpp",
+    ]),
```

Comment:
IMHO this also needs fine granularity: static_transform_broadcaster_node hdrs and srcs should go to a separate library for the moment like in CMake files. Do you need dynamic composition at the moment? At the moment this plugin-like system is not supported at the moment in this repo but we can add it if needed.

Reply:
I removed the nodes, only the tf2_ros library is left.